src/chain.ts

```diff
@@ -50,7 +51,7 @@ export const chain = (
         throw new Error(`Unexpected data on CONNECT: ${head.length} bytes`);
     }
 
-    const { proxyChainId } = sourceSocket as unknown as { proxyChainId: unknown };
+    const { proxyChainId } = sourceSocket as Socket;
```

Comment:
what about using the type directly in `ChainOpts`? we might need to make the `id` optional to allow passing raw `net.Socket` instance, but that seems totally fine